analyzer/codechecker_analyzer/cmd/parse.py

```diff
@@ -392,7 +392,22 @@ def get_output_file_path(default_file_name: str) -> Optional[str]:
 
     for dir_path, file_paths in report_file.analyzer_result_files(args.input):
         metadata = get_metadata(dir_path)
-        for file_path in file_paths:
+
+        specified_file_paths = None
+        if metadata and 'files' in args:
+            # Mapping plists when files are specified to speed up parsing
+            specified_file_paths = [
+                key for key, val in
+                metadata['tools'][0]['result_source_files'].items()
+                if val in args.files
+            ] if 'tools' in metadata \
+                and len(metadata['tools']) > 0 \
+                and 'result_source_files' in metadata['tools'][0] \
+                else None
```

Comment:
If this `result_source_files` section is missing from `metadata.json` then we could fall back to the slow version by using `file_paths` again instead of giving no result due to this `None` value.